### PR TITLE
Pd 3519 assessment record list filter component

### PR DESCRIPTION
**Pull request by iartemiev**
## What's this PR do?
Adds working filters, sorting and also the StarDisplay component.
The filters are currently single-select, pending the development of our custom multi-select component. However, all of the code is written to support multi-select filtering. Would just need to connect the new component after it's ready...

## Screenshots (if appropriate)
image

## Where should the reviewer start?
1. RecordListFilter.jsx
2. mainActions.js
3. mainReducer.js
4. RecordListHeader.jsx
5. StarDisplay.jsx

## How should this be manually tested?
Navigate to a partner assessment and try filtering and sorting the data. Make sure that the Stars/score is showing up.

## Any background context you want to provide?
Using mock data in RecordListFilter.jsx:8-17. I'll remove it after we have API endpoints to populate the filters with.

### CODE REVIEW

File: gulp/src/partnerassessment/components/Records/RecordItem.jsx [hunk 1]
```diff
@@ -3,16 +3,9 @@ import PropTypes from 'prop-types';
 import {connect} from 'react-redux';
 import {Collapse} from 'react-bootstrap';
 import moment from 'moment';
-
+import StarDisplay from '../../../common/ui/StarDisplay';
```
Comment (aljordan82) [inline gulp/src/partnerassessment/components/Records/RecordItem.jsx]: import StarDisplay from 'common/ui/StarDisplay';

Comment (iartemiev) [inline gulp/src/partnerassessment/components/Records/RecordItem.jsx]: Corrected

File: gulp/src/partnerassessment/components/Records/RecordListHeader.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  render() {
+    const {aggregateScore} = this.props;
+    const recordDate = this.props.recordDate || 'July 20, 2017 - December 20, 2017';
+    return (
+      <div>
+        <div>Review Communication Record</div>
+        <div className="record-list-header">{recordDate}</div>
+        <div>
+          <StarDisplay score={aggregateScore}/>
+          <span className="record-total-score">{aggregateScore}</span>
```
Comment (aljordan82) [inline gulp/src/partnerassessment/components/Records/RecordListHeader.jsx]: add aggregateScore.`toPrecision(2)`

File: gulp/src/partnerassessment/components/Records/RecordListHeader.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import StarDisplay from '../../../common/ui/StarDisplay';
+/* RecordListHeader */
+export default class RecordListHeader extends Component {
+  render() {
+    const {aggregateScore} = this.props;
+    const recordDate = this.props.recordDate || 'July 20, 2017 - December 20, 2017';
```
Comment (iartemiev) [inline gulp/src/partnerassessment/components/Records/RecordListHeader.jsx]: That's yours

**Comment by aljordan82** [comment]
Update the PaginationList component to show the filtered list length instead of the hard-coded string `1x`

File: gulp/src/partnerassessment/components/Records/RecordListHeader.jsx (hunk 3 above)
Comment (aljordan82) [inline gulp/src/partnerassessment/components/Records/RecordListHeader.jsx]: This needs to get its value from redux, I'll take care of it.

File: gulp/src/partnerassessment/actions/mainActions.js [hunk 4]
```diff
@@ -1,12 +1,96 @@
...
+          return;
+        }
+        // different filter functions for arrays and for primitives
+        const filterFn = filterBy.value instanceof Array ? filterArray : filterValue;
+        filtered = filter(filtered, (doc) => filterFn(doc, filterBy));
+      });
+
+      dispatch(updateFilteredRecordList(filtered));
+      dispatch(doSortRecords());
+    } catch (e) {
```
Comment (aljordan82) [inline gulp/src/partnerassessment/actions/mainActions.js]: updateFilteredRecordList is dispatched in doSortRecords, is there a reason for this?

Comment (iartemiev) [inline gulp/src/partnerassessment/actions/mainActions.js]: Well they both update the filteredDocs.
So we filter -> update filteredDocs
Then sort -> update filteredDocs

**Pull request merged**